workers/unity/Packages/io.improbable.gdk.buildsystem/WorkerBuilder.cs

```diff
@@ -273,74 +239,78 @@ public static void Clean()
-                return false;
-            }
-
             Debug.Log(
-                $"Building \"{buildTarget}\" for worker platform: \"{workerType}\", environment: \"{targetEnvironment}\"");
-
+                $"Building \"{buildContext.WorkerType}\" for platform: \"{buildContext.BuildTargetConfig.Target}\", environment: \"{buildContext.BuildEnvironment}\"");
 
-            var workerBuildData = new WorkerBuildData(workerType, buildTarget);
-            var scenes = spatialOSBuildConfiguration.GetScenePathsForWorker(workerType);
+            var workerBuildData = new WorkerBuildData(buildContext.WorkerType, buildContext.BuildTargetConfig.Target);
+            var scenes = spatialOSBuildConfiguration.GetScenePathsForWorker(buildContext.WorkerType);
 
             var buildPlayerOptions = new BuildPlayerOptions
             {
-                options = buildOptions,
-                target = buildTarget,
+                options = buildContext.BuildTargetConfig.Options,
+                target = buildContext.BuildTargetConfig.Target,
                 scenes = scenes,
                 locationPathName = workerBuildData.BuildScratchDirectory
             };
 
-            var result = BuildPipeline.BuildPlayer(buildPlayerOptions);
-            if (result.summary.result != BuildResult.Succeeded)
+            var buildTargetGroup = BuildPipeline.GetBuildTargetGroup(buildContext.BuildTargetConfig.Target);
+            var activeScriptingBackend = PlayerSettings.GetScriptingBackend(buildTargetGroup);
+            if (activeScriptingBackend != buildContext.ScriptingImplementation)
+            {
+                PlayerSettings.SetScriptingBackend(buildTargetGroup, buildContext.ScriptingImplementation);
+            }
+
+            try
             {
-                if (buildTarget == BuildTarget.Android && string.IsNullOrEmpty(EditorPrefs.GetString("AndroidSdkRoot")))
+                currentContext = buildContext;
+                var result = BuildPipeline.BuildPlayer(buildPlayerOptions);
+                if (result.summary.result != BuildResult.Succeeded)
                 {
-                    Debug.LogWarning($"Unable to build worker {workerType} for platform Android. " +
-                        $"Ensure you have the Android SDK set inside the Unity Editor Preferences.");
-                    return false;
-                }
+                    if (buildContext.BuildTargetConfig.Target == BuildTarget.Android &&
+                        string.IsNullOrEmpty(EditorPrefs.GetString("AndroidSdkRoot")))
+                    {
+                        Debug.LogWarning($"Unable to build worker {buildContext.WorkerType} for platform Android. " +
+                            $"Ensure you have the Android SDK set inside the Unity Editor Preferences.");
+                        return false;
+                    }
```

Comment:
Couldn't this be done before `BuildPlayer`?

Reply:
I have no idea what this is even for. Since Unity will just tell you exactly this I assume?

Reply:
You would think so 🤔